### PR TITLE
Multiplayer screen

### PR DESCRIPTION
**Pull request by DrabWeb**
The basic multiplayer screen and lobby. Match screen to come in a follow up PR.

Also includes:
* `BreadcrumbControl` design adjustments.
* ~~Refactored `WaveOverlayContainer` to `WaveContainer`.~~ moved to #2383. 
* Restructured the `Multiplayer` folder.

### CODE REVIEW

**Comment by smoogipoo** [comment]
Next time please split out the changes into bite-size PRs. Checking 1500 lines of new code is gonna take quite some time.

**Comment by peppy** [comment]
Could you please PR the wave container changes separately? I'd like to review that on its own before tackling this one.

File: osu.Game/Screens/Multiplayer/Components/DrawableRoom.cs [hunk 1]
```diff
@@ -0,0 +1,340 @@
...
+        public DrawableRoom(Room room)
+        {
+            Room = room;
+
+            RelativeSizeAxes = Axes.X;
+            Height = height + SELECTION_BORDER_WIDTH * 2;
+            Masking = true;
+            CornerRadius = 5 + SELECTION_BORDER_WIDTH / 2;
+
+            Children = new Drawable[]
```
Comment (peppy) [inline osu.Game/Screens/Multiplayer/Components/DrawableRoom.cs]: this needs to be moved to the `load()` method if possible

File: osu.Game/Screens/Multiplayer/Components/DrawableRoom.cs [hunk 2]
```diff
@@ -0,0 +1,340 @@
...
+                Action?.Invoke(this);
+            return true;
+        }
+
+        private void displayName(string value)
+        {
+            name.Text = value;
+        }
+
+        private void displayUser(User value)
```
Comment (peppy) [inline osu.Game/Screens/Multiplayer/Components/DrawableRoom.cs]: any reason for having these methods if they are only used in one place?

File: osu.Game/Screens/Multiplayer/Components/RoomInspector.cs [hunk 3]
```diff
@@ -299,6 +323,36 @@ protected override void UpdateAfterChildren()
             participantsScroll.Height = DrawHeight - topFlow.DrawHeight;
         }
 
+        private void nullState()
```
Comment (peppy) [inline osu.Game/Screens/Multiplayer/Components/RoomInspector.cs]: i don't like these methods. instead, make an `updateState` method which performs *all* state changes and call it from the setter and anywhere else a state update is required. see `osu.Game.Overlays.BeatmapSet.Info` and all other overlay relevant classes for examples to follow.

File: osu.Game/Screens/Multiplayer/Components/DrawableRoom.cs (hunk 1 above)
Comment (DrabWeb) [reply osu.Game/Screens/Multiplayer/Components/DrawableRoom.cs]: I think these changes on `DrawableRoom` should be done in a separate PR, as it is not really a part of this one apart from being moved to a new folder. They should probably also be done in `RoomInspector`.

**Comment by DrabWeb** [comment]
Closing this in favour of splitting into multiple PRs. This one got way out of hand and was not intended to be so big.

**Pull request closed without merging**